### PR TITLE
overlay: Remove /boot RW workaround for kdump

### PR DESCRIPTION
**Pull request by travier**
Remove workaround keeping `/boot` read only when kdump support was
requested now that kdump writes the temporary initrd in
`/var/lib/kdump`.

This reverts 3167d83 12kdump: Remount /boot rw before `kdump.service`.

### CODE REVIEW

**Comment by travier** [comment]
I will do manual testing on FCOS (& RHCOS) to make sure this is OK as the kdump test is currently disabled. (https://github.com/coreos/fedora-coreos-tracker/issues/860)

**Review by jlebon — DISMISSED** [review]
Shouldn't the commit message say "keeping `/boot` read-write"?

**Comment by travier** [comment]
Marking as draft as we need a backport for F34 first: https://bugzilla.redhat.com/show_bug.cgi?id=2003780

**Comment by dustymabe** [comment]
> Marking as draft as we need a backport for F34 first: https://bugzilla.redhat.com/show_bug.cgi?id=2003780

if it's a lot of effort we could just wait for f35 (which is coming pretty soon anyway).

**Comment by dustymabe** [comment]
if you wanted to go ahead and apply this to only f35 now you could move the files into a post script in the manifests (conditional on f35+) and then change the test to use denylist instead and only skip it on f34. We've done this in other places.

**Comment by travier** [comment]
> > Marking as draft as we need a backport for F34 first: https://bugzilla.redhat.com/show_bug.cgi?id=2003780
> 
> if it's a lot of effort we could just wait for f35 (which is coming pretty soon anyway).

I don't think it's a lot of effort but we need the kexec-tools team to make it happen.

> if you wanted to go ahead and apply this to only f35 now you could move the files into a post script in the manifests (conditional on f35+) and then change the test to use denylist instead and only skip it on f34. We've done this in other places.

I think that's more efforts for not much gain. I'll wait as this is not blocking RHCOS.

**Comment by travier** [comment]
https://github.com/coreos/fedora-coreos-config/commit/6cde71e15ec923884d6cbfa83b32ff213d64b124 > This one includes the fixed ~kdump~ `kexec-tools` package.

File: tests/kola/kdump/test.sh [hunk 1]
```diff
@@ -1,12 +1,9 @@
...
 set -xeuo pipefail
 # https://docs.fedoraproject.org/en-US/fedora-coreos/debugging-kernel-crashes/
-# kola: {"minMemory": 4096, "tags": "skip-base-checks"}
-
-# ===== FIXME: Disabled due to broken CI
-echo "Test disabled"
-exit 0
-# =====
+# Only run on QEMU and x86_64 & aarch64 for now:
+# https://github.com/coreos/fedora-coreos-tracker/issues/860
```
Comment (dustymabe) [inline tests/kola/kdump/test.sh:5]: Hmm. I thought the work you were doing here was to close out https://github.com/coreos/fedora-coreos-tracker/issues/860 

Is this test still failing on cloud platforms and needs further investigation there?

Comment (travier) [reply tests/kola/kdump/test.sh:5]: This is the first step: get it to work on QEMU in CI.
I'll work on investigating issues in cloud platforms as a second step.

**Comment by travier** [comment]
```
[2021-10-14T15:23:30.335Z] --- PASS: ext.config.kdump (128.26s)
```

File: tests/kola/kdump/test.sh (hunk 1 above)
Comment (dustymabe) [reply tests/kola/kdump/test.sh:5]: 👍

**Review by dustymabe — APPROVED** [review]
LGTM

**Pull request merged**